src/OpenSSL/crypto.py

```diff
@@ -642,23 +642,17 @@ def __getattr__(self, name):
             _lib.OPENSSL_free(result_buffer[0])
         return result
 
-    def _cmp(op):
-        def f(self, other):
-            if not isinstance(other, X509Name):
-                return NotImplemented
-            result = _lib.X509_NAME_cmp(self._name, other._name)
-            return op(result, 0)
-
-        return f
+    def _cmp(self, other):
+        if not isinstance(other, X509Name):
+            return NotImplemented
 
-    __eq__ = _cmp(__eq__)
-    __ne__ = _cmp(__ne__)
+        return _lib.X509_NAME_cmp(self._name, other._name)
 
-    __lt__ = _cmp(__lt__)
-    __le__ = _cmp(__le__)
+    def __eq__(self, other):
+        return self._cmp(other) == 0
```

Comment:
This isn't quite right -- in the event `other` is not an `X509Name` then this is going to do `NotImplemented == 0` which isn't what we want.

Reply:
whoops, that should of course be raising

Reply:
To be clear: We want to return `NotImplemented` from `__eq__` if the other side isn't an `X509Name` since that ensures that the RHS's `__eq__` can be called.

Reply:
Yeah i just found that out when looking up the difference between NotImplemented and NotImplementedError

I will think about how to solve this

Reply:
I wanted to avoid the code duplication, but i think its not worth it for the 2 isinstance calls.